src/index.js

```diff
@@ -84,25 +84,25 @@ async function runner(ctx, name) {
 
 	try {
 		if (name) write(SUITE(kleur.black(` ${name} `)) + ' ');
-		for (hook of before) await hook(state);
+		for (hook of before) await hook(state, name);
 
 		for (test of arr) {
 			try {
-				for (hook of bEach) await hook(state);
+				for (hook of bEach) await hook(state, `${name} ${test.name}`);
```

Comment:
would need your brilliance here @lukeed on if doing a `.trim()` here would be suffice, and to null check the name here. To not end up with `"undefined something"` for `test()` calls.
``